cdap-data-fabric/src/main/java/co/cask/cdap/data/view/MDSViewStore.java

```diff
@@ -85,70 +107,60 @@ public Boolean apply(ViewMetadataStoreDataset mds) throws Exception {
 
   @Override
   public boolean exists(final Id.Stream.View viewId) {
-    return execute(new TransactionExecutor.Function<ViewMetadataStoreDataset, Boolean>() {
+    return execute(new TxCallable<Boolean>() {
       @Override
-      public Boolean apply(ViewMetadataStoreDataset mds) throws Exception {
-        return mds.exists(getKey(viewId));
+      public Boolean call(DatasetContext context) throws Exception {
+        return getViewDataset(context).exists(getKey(viewId));
       }
     });
   }
 
   @Override
   public void delete(final Id.Stream.View viewId) throws NotFoundException {
-    boolean notFound = execute(new TransactionExecutor.Function<ViewMetadataStoreDataset, Boolean>() {
-      @Override
-      public Boolean apply(ViewMetadataStoreDataset mds) throws Exception {
-        if (!mds.exists(getKey(viewId))) {
-          return true;
+    try {
```

Comment:
The execute help always propagate exceptions as `RuntimeException`, while in here, we want to preserves the `NotFoundException`.